Upstream `defaultAlternateFormControlDesignEnabled` and `defaultVideoFullscreenRequiresElementFullscreen`

hortont424 opened the pull request:
#### c976622eda347ae0d2691521cad71e40d6ae5acf
<pre>
Upstream `defaultAlternateFormControlDesignEnabled` and `defaultVideoFullscreenRequiresElementFullscreen`
<a href="https://bugs.webkit.org/show_bug.cgi?id=257908">https://bugs.webkit.org/show_bug.cgi?id=257908</a>
rdar://110544922

Reviewed by Megan Gardner and Aditya Keerthi.

* Source/WebCore/platform/ios/Device.cpp:
(WebCore::deviceClassIsReality):
* Source/WebCore/platform/ios/Device.h:
* Source/WebKit/Shared/UserInterfaceIdiom.h:
* Source/WebKit/Shared/UserInterfaceIdiom.mm:
(WebKit::currentUserInterfaceIdiomIsSmallScreen):
(WebKit::currentUserInterfaceIdiomIsReality):
(WebKit::currentUserInterfaceIdiom):
(WebKit::setCurrentUserInterfaceIdiom):
(WebKit::updateCurrentUserInterfaceIdiom):
(WebKit::setCurrentUserInterfaceIdiomIsSmallScreen): Deleted.
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::platformInitializeWebProcess):
(WebKit::WebProcessPool::registerNotificationObservers):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::WebProcess::userInterfaceIdiomDidChange):
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/UserInterfaceIdiom.h:
* Source/WebKit/Shared/UserInterfaceIdiom.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
Extend our existing UserInterfaceIdiom mechanism to support the new idiom.

* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultAlternateFormControlDesignEnabled):
(WebKit::defaultVideoFullscreenRequiresElementFullscreen):
Adopt it to implement these two preferences.

Canonical link: <a href="https://commits.webkit.org/265038@main">https://commits.webkit.org/265038@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5de3f31cd41eee61edc74b85153b8ce4e9ee30be

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9582 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9842 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10083 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11243 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/9380 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9593 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11821 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9799 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/12297 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9734 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/10596 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/8169 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11402 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/7891 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/8719 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/16127 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/8986 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/8867 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12199 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9355 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/7602 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/8550 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/12771 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1096 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9127 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/Shared/UserInterfaceIdiom.h:
```diff
@@ -27,12 +27,35 @@
+        WebKit::UserInterfaceIdiom::Reality
+    >;
+};
+}
```
Can we `serialization.in`?

hortont424 replied:
Maybe? I have to see what the caller is I guess.

hortont424 committed:
Upstream `defaultAlternateFormControlDesignEnabled` and `defaultVideoFullscreenRequiresElementFullscreen`
https://bugs.webkit.org/show_bug.cgi?id=257908
rdar://110544922

Reviewed by Megan Gardner and Aditya Keerthi.

* Source/WebCore/platform/ios/Device.cpp:
(WebCore::deviceClassIsReality):
* Source/WebCore/platform/ios/Device.h:
* Source/WebKit/Shared/UserInterfaceIdiom.h:
* Source/WebKit/Shared/UserInterfaceIdiom.mm:
(WebKit::currentUserInterfaceIdiomIsSmallScreen):
(WebKit::currentUserInterfaceIdiomIsReality):
(WebKit::currentUserInterfaceIdiom):
(WebKit::setCurrentUserInterfaceIdiom):
(WebKit::updateCurrentUserInterfaceIdiom):
(WebKit::setCurrentUserInterfaceIdiomIsSmallScreen): Deleted.
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::platformInitializeWebProcess):
(WebKit::WebProcessPool::registerNotificationObservers):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::WebProcess::userInterfaceIdiomDidChange):
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/UserInterfaceIdiom.h:
* Source/WebKit/Shared/UserInterfaceIdiom.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
Extend our existing UserInterfaceIdiom mechanism to support the new idiom.

* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultAlternateFormControlDesignEnabled):
(WebKit::defaultVideoFullscreenRequiresElementFullscreen):
Adopt it to implement these two preferences.

Canonical link: https://commits.webkit.org/265038@main
